File: C++_Doc/4_EstructurasDeControl.cpp
/*
 *  Tercer Archivo
* Nombre: Tipos De Estructuras De Datos.cpp
* Autor: ByEmmanuel
* Fecha: 05-09-2024
* Descripcion: Tipos de Estructuras de control en C++
* Estas estructuras de controlo definen para donde debe ir el rumbo del programa
*/

#include <iostream>

int estructurasDeControl(){
    // Estructura de control if
    int numero = 5;
    if (numero > 0){
        std::cout << "El numero es positivo" << std::endl;
    }
    else if (numero < 0){
        std::cout << "El numero es negativo" << std::endl;
    }
    else{
        std::cout << "El numero es cero" << std::endl;
    }

    // Estructura de control switch
    int dia = 3;
    switch (dia){
        case 1:
            std::cout << "Lunes" << std::endl;
            break;
        case 2:
            std::cout << "Martes" << std::endl;
            break;
        case 3:
            std::cout << "Miercoles" << std::endl;
            break;
        case 4:
            std::cout << "Jueves" << std::endl;
            break;
        case 5:
            std::cout << "Viernes" << std::endl;
            break;
        case 6:
            std::cout << "Sabado" << std::endl;
            break;
        case 7:
            std::cout << "Domingo" << std::endl;
            break;
        default:
            std::cout << "Dia invalido" << std::endl;
            break;
    }

    // Estructura de control while
    int i = 0;
    while (i < 5){
        std::cout << i << std::endl;
        i++;
    }

    // Estructura de control do-while
    int j = 0;
    do{
        std::cout << j << std::endl;
        j++;
    } while (j < 5);

    // Estructura de control for
    for (int k = 0; k < 5; k++){
        std::cout << k << std::endl;
    }

    return 0;
}

File: Java_Doc/ONE_Introduccion_A_Java/src/main/java/ONE_Introduccion_A_Java/_4_EstructurasDeControl.java
package src.main.java.ONE_Introduccion_A_Java;

/*
 *  Cuarto Archivo
 * Nombre: Tipos De Estructuras De Control
 * Autor: ByEmmanuel
 * Fecha: 20-09-2024
 * Descripcion: Tipos de Estructuras de control en Java
 * Estas estructuras de control definen para donde debe ir el rumbo del programa
 *
 */


public class _4_EstructurasDeControl {

    //Estructuras de control en Java

    //Definimos una variable global para el documento pero solo visible para el mismo el cual sera un array y le pasare slos valores mediante acceso a su indice
    private static final int[] arrayList = {1,2,3,4,5,6,7,8,9};

    public static void main(String[] args) {

        estructura_If(arrayList[0], arrayList[4]);
        estructura_SwitchCase(arrayList);
        estructura_For(arrayList);
        estructura_While();
        estructura_TryCatch();

    }



    //Si le pongo return el codigo hasta ahi se detiene
    //En cambio si no le pongo return se ejecutara todo bloque de codigo
    private static int estructura_If(int num1, int num2){

        //Estructura de control If

        if(num1 == num2){
            //Esto no se ejecuta
            System.out.println("Es verdadero");
            return 1;
        }else{
            if (num1 < num2){
                System.out.println("El numero es mayor");
                return 2;
            }
            if (num1 + 4 == num2){
                System.out.println("El numero es igual");
                return 3;
            }
            System.out.println("El nummero no es igual ");
            return 0;
        }

        
    }


    private static void estructura_SwitchCase(int[] array) {
    
        //Estructura de control Switch Case Que ejecutara el 4to caso

        switch (array[3]){
            case 1:
                System.out.println("El numero es 1");
                break;
            case 2:
                System.out.println("El numero es 2");
                break;
            case 3:
                System.out.println("El numero es 3");
                break;
            case 4:
                System.out.println("El numero es 4");
                break;
            default:
                System.out.println("El numero no es ninguno de los anteriores");
                break;
        }


    }

    private static void estructura_For(int [] array) {

        //Estructura de control For

        for (int i = 0; i < array.length; i++){
            System.out.println("El valor del array en la posicion " + i + " es: " + array[i]);
        }

        //For each 

        for (int i : array) {
            System.out.println("El valor del array es: " + i);
        }

    }

    private static void estructura_While() {
    
        //Estructura de control While

        int i = 0;

        // while (i < arrayList.length){
        //     System.out.println("El valor del array en la posicion " + i + " es: " + arrayList[i]);
        //     i++;
        // }

        do {
            System.out.println("El valor del array en la pos " + i + " es: " + arrayList[i]);
            i++;
        } while (i < arrayList.length);

    }



    private static void estructura_TryCatch(){

        //Estructura de control Try Catch

        try {
            int num = 10/0;
            System.out.println("El resultado de la division es: " + num);
        } catch (Exception e) {
            System.out.println("No se puede dividir por 0");
        }finally{
            System.out.println("Esto se ejecutara siempre");
        }

    }

}
